src/config/config.swift

```diff
@@ -0,0 +1,179 @@
+    } catch {
+      throw FcitxCodingError.innerError(path: pathPrefix, context: json, error: error)
+    }
+  }
+  // Group
+  var children: [Config] = []
+  for (key, subJson): (String, JSON) in json {
+    if key == "Value" || key == "Description" || key == "DefaultValue" || key == "Type" {
+      continue
+    }
+    do {
+      children.append(try parseJSON(subJson, pathPrefix + "/" + key))
+    } catch {
+      throw FcitxCodingError.innerError(
+        path: pathPrefix + "/" + key, context: subJson, error: error)
+    }
+  }
+  return Config(path: pathPrefix, description: description, kind: .group(children))
+}
+
+func parseOptionJSON(_ json: JSON, _ type: String) throws -> any Option {
+  if type == "Integer" {
+    return try IntegerOption.decode(json: json)
+  } else if type == "Boolean" {
+    return try BooleanOption.decode(json: json)
+  } else if type == "String" {
+    return try StringOption.decode(json: json)
+  } else if type == "Enum" {
+    return try EnumOption.decode(json: json)
+  } else if type == "Key" {
+    return try KeyOption.decode(json: json)
+  } else if type == "List|String" {
+    return try ListOption<String>.decode(json: json)
+  } else if type == "List|Key" {
+    // TODO
+    return try ListOption<String>.decode(json: json)
+  } else {
+    return try UnknownOption.decode(json: json)
+  }
+}
+
+enum FcitxCodingError: Error {
+  case innerError(path: String, context: JSON, error: any Error)
+  case invalidArgument(context: JSON)
+}
+
+protocol FcitxCodable {
+  static func decode(json: JSON) throws -> Self
+  static func decode(_ str: String) throws -> Self
+  // TODO: func encode() -> String
+}
+
+extension FcitxCodable {
+  static func decode(_ str: String) throws -> Self {
+    return try Self.decode(json: JSON(parseJSON: str))
+  }
+}
+
+extension Int: FcitxCodable {
+  static func decode(json: JSON) throws -> Int {
+    // json is like "100"
+    if let int = Int(json.stringValue.trimmingCharacters(in: CharacterSet(charactersIn: "\""))) {
+      return int
+    } else {
+      throw FcitxCodingError.invalidArgument(context: json)
+    }
+  }
+  func encode() -> String {
+    String(self)
+  }
+}
+
+extension Bool: FcitxCodable {
+  static func decode(json: JSON) throws -> Self {
+    if json.stringValue == "True" {
+      return true
+    } else if json.stringValue == "False" {
+      return false
+    } else {
+      throw FcitxCodingError.invalidArgument(context: json)
+    }
+  }
+  func encode() -> String {
+    if self { return "True" } else { return "False" }
+  }
+}
+
+extension String: FcitxCodable {
+  static func decode(json: JSON) throws -> Self {
+    if let decoded = json.string {
+      return decoded
+    } else {
+      throw FcitxCodingError.invalidArgument(context: json)
+    }
+  }
+  func encode() -> String {
+    let encodedString = self.replacingOccurrences(of: "\"", with: "\\\"")
+      .replacingOccurrences(of: "\\", with: "\\\\")
```

Comment:
This will change `"` to `\\"`, is that expected?

Reply:
No. Good catch!